### PR TITLE
feat(plugins): send accept header in webServer url checking

### PR DESCRIPTION
**Pull request by ghost91-**
This PR sets the Accept HTTP header to `*/*` for HTTP requests made by the webServer plugin when using a URL to check if the server has already started. So far, the Accept header was not being sent at all, which should be semantically equivalent to sending `*/*`, according to [RFC 7231](https://www.rfc-editor.org/rfc/rfc7231#section-5.3.2). However, there are some web servers that handle this differently. Most notably, [Vite](https://vitejs.dev/)'s server just returns a 404 if the `Accept` header is missing from the request. I created https://github.com/vitejs/vite/issues/9520 in order to address the issue on their side, but I think it's also fine to just add the header here, which is why I created this PR.

### CODE REVIEW

**Comment by ghost91-** [comment]
The test failures seem to be completely unrelated, from what I can tell.

Is there anything I can do to help this PR to get a review?

File: tests/playwright-test/web-server.spec.ts
Comment (rwoll) [inline tests/playwright-test/web-server.spec.ts:438]: There's a built-in `server` fixture for testing, so that should be used instead of creating a server inline here. See https://github.com/microsoft/playwright/blob/9795a4ba60479fc75de7187e681ba9e90891bb54/tests/page/page-goto.spec.ts#L72 for an example.

**Review by rwoll — CHANGES_REQUESTED** [review]
Thanks for the PR. Minor change needed, but otherwise LGTM.

The CI failures are unrelated. (If you re-base—although not necessary—they should go away.)

File: tests/playwright-test/web-server.spec.ts
Comment (ghost91-) [reply tests/playwright-test/web-server.spec.ts:438]: Thanks for the feedback, I'll adjust it accordingly. However, I just copied it from other tests in the same file. Should the other tests also be adjusted to use the `server` fixture`?

**Review by rwoll — APPROVED** [review]

File: tests/playwright-test/web-server.spec.ts
Comment (rwoll) [reply tests/playwright-test/web-server.spec.ts]: Oops—thanks for pointing that out.

> Should the other tests also be adjusted to use the server fixture?

I'll have to take a closer look before cleaning those up, but I've merged your change.

**Pull request merged**
